interpreter/cling/lib/Interpreter/IncrementalJIT.cpp

```diff
@@ -46,11 +48,29 @@ IncrementalJIT::IncrementalJIT(
   Builder.setExecutorProcessControl(std::move(EPC));
 
   // FIXME: In LLVM 13 this only works for ELF and MachO platforms
-  // Builder.setObjectLinkingLayerCreator(
-  //     [&](ExecutionSession &ES, const Triple &TT) {
-  //       return std::make_unique<ObjectLinkingLayer>(
-  //           ES, std::make_unique<jitlink::InProcessMemoryManager>());
-  //     });
+#if !defined(_WIN32)
+  if (std::getenv("CLING_JITLINK") || std::getenv("CLING_JITLINK_DEBUG")) {
+    Builder.setObjectLinkingLayerCreator([&](ExecutionSession& ES,
+                                             const Triple& TT) {
+      auto GetMemMgr = []() {
+        return std::make_unique<SectionMemoryManager>();
+      };
+      auto ObjLinkingLayer =
+          std::make_unique<RTDyldObjectLinkingLayer>(ES, std::move(GetMemMgr));
```

Comment:
@SimeonEhrig @vgvassilev This option is highly confusing and does not implement what the commit message says: `RTDyldObjectLinkingLayer` is RuntimeDyld, *not* JITLink! Can we please remove this option or implement it properly?

Reply:
Does it means, JITLink is not required to register a GDB listener and add source code debug information to jitted code?

Reply:
No, I think this also works with RuntimeDyld in principle, if you register the right plugins